### DISCUSSION TITLE
Teslamate running not in docker - how to backup?

### DISCUSSION THREAD

**Discussion by snu-snu-snu** [teslamate-org/teslamate#3248] [Q&A] [answered]
Hi,
I am total noob in linux but I managed to get teslamate running on raspberry without docker (do not know why I decided that... but there...

Now I want to backup all Teslamate (as I will be changing drives/cards).
I am trying to backup Teslamate. the command line for this backup in documentation is:
`docker compose exec -T database pg_dump -U teslamate teslamate > /backuplocation/teslamate.bck`

the documentation shows how to install it in docker, manual in linux, etc. but the backup option is for docker only
I would use the /var/ so my command would be, if I run it in docker:
`docker compose exec -T database pg_dump -U teslamate teslamate > /var/teslamate.bck`

but I cannot get this work.
what am I doing wrong? I understand that command should be different...  where is my error?

any help really appreciated

**Comment by cwanja** [comment]
If it's not running within Docker, none of those commands will work. Can you run the postgresql commands natively? 

https://www.postgresql.org/docs/current/backup.html

**Comment by DrMichael** [comment]
> `docker compose exec -T database pg_dump -U teslamate teslamate > /var/teslamate.bck`

It should probably be `docker-compose`. And make sure /var/teslamate.bck is writable.

**Reply by init6-git** [reply]
He's not running docker so needs to just be the pg_dump command.

**Comment by miaszh** [comment]
Could anyone backup a db without docker?
I've installed manually on a machine running linux (without docker).
How can I backup my data to be able to import it later to a docker version?

Any help appreciated!

**Reply by cwanja** [reply]
https://www.postgresql.org/docs/current/backup.html

**Reply by miaszh** [reply]
Hi
I tried.. but still no luck!
`pg_dumpall: error: could not connect to database "template1": connection to server on socket "/var/run/postgresql/.s.PGSQL.5432" failed: FATAL:  role "root" does not exist
`

**Reply by cwanja** [reply]
How did you install TeslaMate?

**Reply by miaszh** [reply]
a horrible mistake, manually on a running linux laptop, so I want to backup my data to migrate to a more stable docker install.
Do I have any chance?

**Reply by cwanja** [reply]
What database technology are you using? If you followed the [manual install docs](https://docs.teslamate.org/docs/installation/debian), you should be using PostgreSQL. Are you sure the database is template1? Also looks like user / permissions issues.

**Reply by miaszh** [reply]
Yes, I used the manual install docs, and using PostgresSQL

This is the command i tried:
`sudo pg_dumpall > dumpfile`

doesn't look like i entered the "template1" part.

**Reply by DrMichael** [reply]
Try the user, which owns the teslamate database. Usually "teslamate":

`sudo pgdumpall -U teslamate
`

**Reply by miaszh** [reply]
> Try the user, which owns the teslamate database. Usually "teslamate":
> 
> 
> 
> `sudo pgdumpall -U teslamate
> 
> `

Nope 😒 same error "with user "teslamate""

**Reply by jlestel** [reply]
what is your database called? is it “template1”?

Here is my command: 

`pg_dump -h localhost -U teslamate -p 5432 -d teslamate > backup.sql`

**Reply by DrMichael** [reply]
Again, what is the owner of the database?

**Reply by miaszh** [reply, marked as answer]
> what is your database called? is it “template1”?
> 
> 
> 
> Here is my command: 
> 
> 
> 
> `pg_dump -h localhost -U teslamate -p 5432 -d teslamate > backup.sql`

Looks like it created a file finally!!
How should I import it back after a docker installation?
